pkg/podmounter/mountoptions/mount_options.go

```diff
@@ -55,20 +64,30 @@ func Send(ctx context.Context, sockPath string, options Options) error {
 
 var (
 	messageRecvSize = 1024
-	// We only pass one file descriptor and its 32 bits
+	// We only pass one file descriptor and it's 32 bits
 	unixRightsRecvSize = syscall.CmsgSpace(4)
 )
 
 // Recv receives passed mount options via `Send` function through given `sockPath`.
 func Recv(ctx context.Context, sockPath string) (Options, error) {
 	warnAboutLongUnixSocketPath(sockPath)
 
-	l, err := net.Listen("unix", sockPath)
+	var lc net.ListenConfig
+	l, err := lc.Listen(ctx, "unix", sockPath)
 	if err != nil {
 		return Options{}, fmt.Errorf("failed to listen unix socket %s: %w", sockPath, err)
 	}
 	defer l.Close()
 
+	// `l.Accept` does not respect `ctx`'s deadline, we need to call `ul.SetDeadline` to ensure `l.Accept` has a deadline.
+	if deadline, ok := ctx.Deadline(); ok {
```

Comment:
Is it useful to make this a function? It's repeated twice here and I'm not sure if we'd want to use this more often

Reply:
So seems like these are two different types in two functions, `net.UnixListener` and `net.UnixConn` and they don't share any common interface. So we need to have a generic function with a type bound on `SetDeadline` method. Not sure if this is better or create confusion:
```go
type deadline interface {
	SetDeadline(t time.Time) error
}

func populateDeadlineFromContext[Conn deadline](ctx context.Context, conn Conn) error {
	if deadline, ok := ctx.Deadline(); ok {
		return conn.SetDeadline(deadline)
	}

	return nil
}
```